hw07.c

```diff
@@ -0,0 +1,246 @@
+//typedef TGAImage* (*func_array[])(TGAImage *, size_t, size_t, RGBA *);
+
+void watch_draw_time(TGAImage* self, const int hours, const int minutes){
+
+    FILE *file = fopen("aw.tga","wb");
+    assert(file);
+
+    int firstH = hours / 10;;
+    int secondH = hours % 10;
+    int firstMin = minutes / 10;
+    int secondMin = minutes % 10;
+
+    RGBA fg = {.red = 30, .green = 10, .blue = 10, .alpha = 255};
+//zde musem volat fci s fcnim ukazatel na pole a pdle first/,... hodnot se zavola rovnou dana fce,,nepamatujeu se uz inicializaci atd
+//POLE FUNKCNICHU UKAZATELU
+    void (*func_array[])(TGAImage*, int, int, RGBA*) = {
+        draw_zero, draw_one, /*draw_two, draw_three, draw_foure, draw_five, draw_six, draw seven, draw_eight, draw_nine*/
+    };
+//jen budu omezovat nebo pricit konec zacatek k ouradnicim(prevod do 1D to pak prepocita na ta mista v celem poli, ktere je uz predpripraveno)
+    //draw_one(self,0,0,&fg);
+    /*void (*fptr)(TGAImage*, int, int, RGBA*);
+    fptr = &draw_one;*/
+
+   /* func_array[firstH](self, 0, 0, &fg);
+    func_array[secondH](self, 184, 0, &fg);
+    func_array[firstMin](self, 0, 224, &fg);
+    func_array[secondMin](self, 184, 224, &fg);*/
+
+    int writeCount = fwrite( &self->header, sizeof (TGAheader),1,file); //velikost prvku je header celej a je jen jeden
+    assert(writeCount == 1);
+
+    writeCount = fwrite(&self->data, sizeof(RGBA), tga_width(self) * tga_height(self) * sizeof(RGBA),file);
+
+    fclose(file);
+}
+
+
+//POTREBUJU INIT BRIGHTNESS NENI TO pro cernobile nebo c oto kurva je
+TGAImage *tga_new(const int height, const int width){
+    TGAImage *tga = (TGAImage*)malloc(sizeof(TGAImage));// na heapu takze to neni jen loklani na stacku
+  //  ALLOCTEST( TGAImage );
+
+    memset(&tga->header, 0, sizeof(TGAheader)); //potrebuje adresu ukazatele na novy objekt tga, pac fce to chce
+//TADY CHYBI NEKERE HODNOTY
+    //tga->header == (*tga).header
+    tga->header.image_type = 2;
+    tga->header.depth = 32;
+    tga->header.descriptor |= 1 << 3;
+    tga->header.descriptor |= 1 << 5;
+
+
+    //zapis rozmeru na dva bytes, prvni vyssii jsou nuly, pak cislo 0-255 zapsane binarne v nizsim bytu
+    memcpy(&tga->header.width, &width, 2 );
+    memcpy(&tga->header.height, &height, 2 );
+
+    //ted alokuje pamet pro pixely podle = velikost struct pixel x vyska x sirka
+    tga->data = (RGBA*)malloc(sizeof(RGBA) * width * height); //data je typu RGBA*
+    //nastavime celou alokovanou  pamet pro data
+    memset(tga->data, 0, sizeof(RGBA) * width * height); //tga->data uz je ukazatel
+
+    return tga;
+
+}
+
+void tga_free(TGAImage **self){
+
+    free((*self)->data);
+    (*self)->data = NULL;
+    free(*self);
+    *self = NULL;
+
+}
+
+
+int main(int argc, char **argv){
+    int hours = 0;
+    int minutes = 0;
+
+    if (argc != 3)
+    {
+        return 1;
+    }
+    else
+    {
+        hours = strtol(argv[1],NULL,10);
```

Comment:
Tohle predpoklada long int.
Ty chces: https://www.tutorialspoint.com/c_standard_library/c_function_atoi.htm

```suggestion
        hours = atoi(argv[1]);
```